### ISSUE TITLE
Version 0.3.8 stuck (runs forever)

### ISSUE THREAD

**Issue by barakman** [OpenZeppelin/solidity-docgen#75]
Specifically, in file `docgen.js`, the following line never returns:
```
yield fs_extra_1.default.outputFile(dest, page.render(templates.prelude));
```
More specifically, the following part of the line above never returns:
```
const x = page.render(templates.prelude);
```
Tested environment:
- Windows 10
- node 10.16.0
- npm 6.9.0

**Comment by barakman** [comment]
Update:

This problem can be tracked down to file `sitemap.js`, in the following line:
```
for (const target of c.linkable) {
```
It appears that any attempt to refer to `c.linkable` halts the execution.

I tried `console.log(Object.keys(c));` and got `[ 'source', 'file', 'astNode' ]`.

Do `c.linkable` should be `undefined` as far as my understanding of Javascript goes.

I tried `console.log(c.linkable);` and even `console.log(typeof c.linkable);`, but the program halts...

**Comment by barakman** [comment]
**Update 2:**

OK, I get it, `linkable` is not an object attribute, but a getter function (in `class SolidityContract`).

The "freeze" occurs on any attempt to refer to `this.functions`.

**Comment by frangio** [comment]
Thanks again for reporting this with so much detail! And thanks for the patience. :pray: 

I can reproduce the issue, I'll look into it.

**Comment by frangio** [comment]
This bug was introduced in https://github.com/OpenZeppelin/solidity-docgen/commit/031c9a09130978ce66c2361a1ef21e0e9beb9767 because a RegExp was changed in a way that made it also match the empty string, which made looping over matches an infinit loop. Fixed in https://github.com/OpenZeppelin/solidity-docgen/commit/39c9e9b51eb049b6e222a961740147b4bbde692a and released as 0.3.9.

I don't understand  why I didn't run into this with OpenZeppelin's code...
